### ISSUE TITLE
2.1.0 doesn't define make() ?

### ISSUE THREAD

**Issue by psnider** [senecajs/seneca#453]
I just upgraded Seneca to the latest version, 2.1.0,
and my code stopped working.
After discovering that my code complains that its seneca instance doesn't have the make() function, I followed the documentation:
http://senecajs.org/tutorials/understanding-data-entities.html

and found that a fresh install also fails in the same way.
I'm using npm v3.3.9, and node v5.0.0:
```
cd /tmp
mkdir test
cd test
npm install seneca
node
```
Then from within the node REPL:
```
var seneca = require('seneca')()
var foo = seneca.make('foo')
```
results in:
> TypeError: seneca.make is not a function

What's changed?
And what should I do?

Thanks

**Comment by mihaidma** [comment]
starting from seneca 2.0 the seneca-entity was extracted as a separate plugin, you have to explicitly do a `use('seneca-entity')`.
A sample: https://github.com/senecajs/seneca-postgres-store/blob/master/test/postgres.test.js#L29

**Comment by mcdonnelldean** [comment]
@psnider Yup what @mihaidma says.

We are refreshing the instructions for Seneca in the next week. Going forward there will an explicit upgrade document for each breaking change. It is currently being worked on.

**Comment by psnider** [comment]
After I added:
> seneca.use('entity')

things worked better, but now I get the rather cryptic message:
> ./node_modules/seneca-transport/lib/transport-utils.js:492
      raw[key] = seneca.make$(value)

Is this a bug, or am I missing something else related to the 2.1 changes?

**Comment by mihaidma** [comment]
Could you please do a `npm ls | grep seneca` and post the seneca version, entity version, transport version?

How can we reproduce? What's your app structure?

**Comment by mcdonnelldean** [comment]
I'm pretty sure this is due to transport needing entities. Its fixed in 3.0 which is going out in about a week. Could you follow Mihai's instructions above and also show your load order?

I have a feeling entites are loading late.

Originally entities had an odd dependency on entities but it's solved in later versions

Kindest Regards,

Dean

On 19 Jul 2016, at 21:37, Mihai Dima <notifications@github.com<mailto:notifications@github.com>> wrote:


Could you please do a npm ls | grep seneca and post the seneca version, entity version, transport version?

How can we reproduce? What's your app structure?

-

**Comment by psnider** [comment]
npm ls | grep seneca:
```
├─┬ seneca@2.1.0
│ ├─┬ seneca-basic@0.4.0
│ ├── seneca-cluster@0.0.1
│ ├─┬ seneca-mem-store@0.5.0
│ ├── seneca-parambulator@0.2.0
│ ├── seneca-repl@0.2.0
│ ├─┬ seneca-transport@1.2.0
│ ├─┬ seneca-web@0.7.1
└─┬ seneca-entity@1.2.0
```
Reducing my app, I'm still able to reproduce the transport-utils.js:492 error with the following two files:
**people-app.js**:
```
function api(options) {
    // restrict the space of user input actions to those that are public
    var valid_actions = { create: 'create', read: 'read', update: 'update', delete: 'delete' };
    this.add('role:api,path:people', function (msg, respond) {
        var action = valid_actions[msg.action];
        if (action) {
            this.act('role:people', {
                action: action,
                person: msg.person
            }, respond);
        }
        else {
            respond(new Error('unknown action=' + msg.action));
        }
    });
    this.add('init:api', function (msg, respond) {
        this.act('role:web', {
            use: {
                prefix: '/api',
                pin: 'role:api,path:*',
                map: {
                    people: { GET: true, POST: true }
                }
            }
        }, respond);
    });
}

var seneca = require('seneca')({})
seneca.use(api)
seneca.client({type:'tcp', pin:'role:people'})

var express = require('express')
var app = express()
app.use(require('body-parser').json())
app.use(seneca.export('web'))
app.listen(3210)
```

and **people-pin-service.js**:
```
function people(options) {
    this.add('init:people', init);
    function init(msg, done) {
        done();
    }
    this.add('role:people,action:create', function (msg, done) {
        var person = msg.person;
        this.make('person').save$(person, function (error, created_person) {
            if (error) {
                done(error);
            }
            else {
                var response = { person: created_person };
                done(null, response);
            }
        });
    });
    return 'people'
}

var seneca = require( 'seneca' )({})
seneca.use('entity')
seneca.use(people)
seneca.listen({type:'tcp', pin:'role:people'})
```

I run these in two separate shells using:
```
node people-app.js --seneca.log=plugin:web,plugin:api
```
and

```
node people-pin-service.js --seneca.log=plugin:people
```

then curl a create request at it:
```
curl -H "Content-Type: application/json" -X POST -d '{"action":"create", "person":{"name":{"given":"Sally","family":"Smith"}}}' http://localhost:3210/api/people
```

**Comment by mcdonnelldean** [comment]
In the fist one I don't see entity used

Kindest Regards,

Dean

On 19 Jul 2016, at 22:09, psnider <notifications@github.com<mailto:notifications@github.com>> wrote:


npm ls | grep seneca:

??? seneca@2.1.0
? ??? seneca-basic@0.4.0
? ??? seneca-cluster@0.0.1
? ??? seneca-mem-store@0.5.0
? ??? seneca-parambulator@0.2.0
? ??? seneca-repl@0.2.0
? ??? seneca-transport@1.2.0
? ??? seneca-web@0.7.1
??? seneca-entity@1.2.0


Reducing my app, I'm still able to reproduce the transport-utils.js:492 error with the following two files:
people-app.js:

function api(options) {
    // restrict the space of user input actions to those that are public
    var valid_actions = { create: 'create', read: 'read', update: 'update', delete: 'delete' };
    this.add('role:api,path:people', function (msg, respond) {
        var action = valid_actions[msg.action];
        if (action) {
            this.act('role:people', {
                action: action,
                person: msg.person
            }, respond);
        }
        else {
            respond(new Error('unknown action=' + msg.action));
        }
    });
    this.add('init:api', function (msg, respond) {
        this.act('role:web', {
            use: {
                prefix: '/api',
                pin: 'role:api,path:*',
                map: {
                    people: { GET: true, POST: true }
                }
            }
        }, respond);
    });
}

var seneca = require('seneca')({})
seneca.use(api)
seneca.client({type:'tcp', pin:'role:people'})

var express = require('express')
var app = express()
app.use(require('body-parser').json())
app.use(seneca.export('web'))
app.listen(3210)


and people-pin-service.js:

function people(options) {
    this.add('init:people', init);
    function init(msg, done) {
        done();
    }
    this.add('role:people,action:create', function (msg, done) {
        var person = msg.person;
        this.make('person').save$(person, function (error, created_person) {
            if (error) {
                done(error);
            }
            else {
                var response = { person: created_person };
                done(null, response);
            }
        });
    });
    return 'people'
}

var seneca = require( 'seneca' )({})
seneca.use('entity')
seneca.use(people)
seneca.listen({type:'tcp', pin:'role:people'})


I run these in two separate shells using:

node people-app.js --seneca.log=plugin:web,plugin:api


and

node people-pin-service.js --seneca.log=plugin:people


then curl a create request at it:

curl -H "Content-Type: application/json" -X POST -d '{"action":"create", "person":{"name":{"given":"Sally","family":"Smith"}}}' http://localhost:3210/api/people


-

**Comment by psnider** [comment]
The first one, **people-app.js**, only forwards requests to the second one.
And people-app.js doesn't reference any part of the entity package.

So isn't it an error that its referencing entity down in the transport?
I don't understand why I should add:
> seneca.use('entity')

to people-app.js.

**Comment by mcdonnelldean** [comment]
Yes,

But in all instances of seneca you need some form of transport. Since that version of seneca doesn't contain them by default you need to include entities to get transport to work.

As mentioned above, this is resolved in 3.0 as transport no longer has a dependency on it.

Kindest Regards,

Dean

On 19 Jul 2016, at 23:30, psnider <notifications@github.com<mailto:notifications@github.com>> wrote:


The first one, people-app.js, only forwards requests to the second one.
And people-app.js doesn't reference any part of the entity package.

So isn't it an error that its referencing entity down in the transport?
I don't understand why I should add:

seneca.use('entity')

to people-app.js.

-
